Sources/CreateAPI/Generator/Generator+Schemas.swift

```diff
@@ -406,10 +406,40 @@ extension Generator {
         }
     }
 
+    private func makeDiscriminator(info: JSONSchemaContext, context: Context) -> Discriminator? {
+        if let discriminator = info.discriminator {
+            var mapping: [String: TypeIdentifier] = [:]
+
+            if let innerMapping = discriminator.mapping {
+                
+                // TODO: Use type name machinery instead
+                mapping = innerMapping.mapValues { value in 
```

Comment:
I think this should work:

```swift
mapping = innerMapping.mapValues { value in
    guard let key = OpenAPI.ComponentKey(rawValue: value) else {
        throw GeneratorError("TBD") // This should never happen
    }
    // This takes care of options.rename.
    // It _doesn't_ take options.entities.include / exclude into the account.
    // Ideally CreateAPI should throw an error (or warning if --strict is disabled) if
    // the type is exlucded (neat-picking).
    guard let name = getTypeName(for: key) else {
        // TODO:
    }
    return .userDefined(name: name)
}
```

Reply:
```swift
guard let key = OpenAPI.ComponentKey(rawValue: value) else {
    throw GeneratorError("TBD") // This should never happen
}
```
 
The `rawValue:` constructor enforces that the provided value matches this regex: `^[a-zA-Z0-9\.\-_]+$`.
At this stage the mapping values are still prefixed with `"#/components/schemas/"`, which does not conform to said regex.

Reply:
You are right. I just tested it and yes, the string value should be the component name (without "#/components/schemas/").

```swift
String(value.dropFirst("#/components/schemas/".count))
```

That's a nice approach for removing the prefix. I would've usually used `replacingOccurrences`, but this is much better.